### PR TITLE
Migrate to GitInfo for embedding git information

### PR DESCRIPTION
**Pull request by Slendy**
Gets rid of the old system that would copy .txt files to the output directory. Also gets rid of the RazorRuntimeCompilation because apparently it isn't needed for hot reload and was only used for development anyways

### CODE REVIEW

File: ProjectLighthouse/Helpers/VersionHelper.cs [hunk 1]
```diff
@@ -12,64 +11,28 @@ static VersionHelper()
     {
         try
         {
-            CommitHash = ResourceHelper.ReadManifestFile("gitVersion.txt");
-            Branch = ResourceHelper.ReadManifestFile("gitBranch.txt");
+            CommitHash = ThisAssembly.Git.Commit;
+            Branch = ThisAssembly.Git.Branch;
             string commitNumber = $"{CommitHash}_{Build}";
             FullRevision = Branch == "main" ? $"r{commitNumber}" : $"{Branch}_r{commitNumber}";
```
Comment (sudokoko) [inline ProjectLighthouse/Helpers/VersionHelper.cs:17]: Is the try/catch still necessary here seeing as these values are constants?

**Review by sudokoko — CHANGES_REQUESTED** [review]

File: ProjectLighthouse/Helpers/VersionHelper.cs (hunk 1 above)
Comment (TorutheRedFox) [reply ProjectLighthouse/Helpers/VersionHelper.cs:17]: > Is the try/catch still necessary here seeing as these values are constants?

yeah it seems the try/catch was entirely there to handle errors caused by missing/malformed txt files

Comment (Slendy) [reply ProjectLighthouse/Helpers/VersionHelper.cs]: Removed the try catch and the entire constructor since it was unnecessary

**Review by sudokoko — APPROVED** [review]

**Review by Zaprit — APPROVED** [review]
LGTM

**Pull request merged**